### PR TITLE
Mutual TLS communication with Global Adapter

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
$subject. 

### Issues

Related to #2106

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by mpmunasinghe — APPROVED** [review]
LGTM

**Review by NipunaPrashan — APPROVED** [review]

File: adapter/internal/ga/client.go [hunk 1]
```diff
@@ -102,6 +105,26 @@ func initConnection() (*grpc.ClientConn, error) {
 	return conn, nil
 }
 
+func generateTLSCredentialsForXdsClient() credentials.TransportCredentials {
+	conf, _ := config.ReadConfigs()
+	certPool := tlsutils.GetTrustedCertPool(conf.Adapter.Truststore.Location)
+	// There is a single private-public key pair for XDS server initialization, as well as for XDS client authentication
+	certificate, err := tlsutils.GetServerCertificate(conf.Adapter.Keystore.PublicKeyLocation,
```
Comment (mpmunasinghe) [inline adapter/internal/ga/client.go:112]: Shouldn't we use two configs for key and certificate?

**Review by Jayanie — APPROVED** [review]

File: adapter/internal/ga/client.go (hunk 1 above)
Comment (VirajSalaka) [reply adapter/internal/ga/client.go:112]: fixed with https://github.com/wso2/product-microgateway/pull/2150/commits/13c2a3b94cd52cef14753f079b0af775a2fd417b

**Review by mpmunasinghe — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20210825.10

**Comment by choreo-cicd** [comment]
[succeeded] prod-deployment : 20210831.1

**Pull request merged**